packages/core/compute-runtime/src/ProcessHandle.ts

```
@@ -227,6 +227,38 @@ export class ProcessHandleImpl<I, O, R> implements ProcessManager.Handle<I, O> {
+              return Effect.runSync(Deferred.succeed(deferred, undefined));
+            // The foreground turn is done once no handler is active and no further turn work is
+            // queued (no pending alarm); remaining hybernation is only background children, which we
+            // intentionally do not wait for.
+            case Process.State.HYBERNATING:
+              return this.#alarmTimer === null ? Effect.runSync(Deferred.succeed(deferred, undefined)) : Effect.void;
+            case Process.State.FAILED:
+              const error = state.exit.pipe(
+                Option.flatMap(Exit.causeOption),
+                Option.map(Cause.pretty),
+                Option.getOrElse(() => 'Process failed with unknown error'),
+              );
```

Biome (2.4.16)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)